src/scripts/core/axis/y-axis.js

```diff
@@ -17,7 +17,7 @@
             /*jshint eqnull:true */
             var options = this.options.yAxis;
             var domain = this.domain;
-            var zeroAnchor = options.zeroAnchor || options.scaling.zeroAnchor;
+            var zeroAnchor = (typeof options.zeroAnchor !== 'undefined') ? options.zeroAnchor : options.scaling.options.zeroAnchor;
```

Comment:
This catches undefineds, but not nulls. I think `options.zeroAnchor != null` may be a better check